app/models/work.rb

```diff
@@ -116,10 +116,11 @@ def publish_test_doi?
 
       def works_by_user_state(user, state)
         works = []
-        if user.admin_collections.count == 0
-          # Just the user's own works by state
-          works += Work.where(created_by_user_id: user, state: state)
-        else
+
+        # The user's own works by state (if any)
+        works += Work.where(created_by_user_id: user, state: state)
```

Comment:
Instead of a `+=` here, could you combine line. 118 and 121 (Since it is now happening every time)
```suggestion
        works = Work.where(created_by_user_id: user, state: state)
```

Reply:
Funny. I tried that first, but turns out `Work.where` returns an `ActiveRelation` that looks like an array but it is not an array. Since it is not an array the code blows up a few lines down when it does `works << work unless already_included`. I could probably come up with a clever way to handle this but I thought the simplest and cleanest was to use a plain array and append to it.

Reply:
Oh!  You just need a `to_a` on the end of the query.